controllers/storageclusterpeer/storageclusterpeer_controller.go

```diff
@@ -18,46 +18,160 @@ package storageclusterpeer
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the StorageClusterPeer object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
-func (r *StorageClusterPeerReconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// TODO(user): your logic here
+func (r *StorageClusterPeerReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	var err error
+	r.ctx = ctx
+	r.log = log.FromContext(ctx, "StorageClient", request)
+	r.log.Info("Reconciling StorageClusterPeer.")
+
+	// Fetch StorageCluster(s)
+	r.storageCluster, err = util.GetStorageClusterInNamespace(r.ctx, r.Client, request.Namespace)
+	if err != nil && k8serrors.IsNotFound(err) {
+		r.log.Error(err, "StorageCluster for StorageClusterPeer found in the same namespace.")
+		return ctrl.Result{}, err
+	}
+
+	// Fetch the StorageClusterPeer instance
+	r.storageClusterPeer = &ocsv1.StorageClusterPeer{}
+	r.storageClusterPeer.Name = request.Name
+	r.storageClusterPeer.Namespace = request.Namespace
+
+	if err = r.get(r.storageClusterPeer); err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.log.Info("StorageClusterPeer resource not found. Ignoring since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		r.log.Error(err, "Failed to get StorageClusterPeer.")
+		return reconcile.Result{}, err
+	}
+
+	result, reconcileErr := r.reconcilePhases()
+
+	// Apply status changes to the StorageClient
+	statusErr := r.Client.Status().Update(ctx, r.storageClusterPeer)
+	if statusErr != nil {
+		r.log.Error(statusErr, "Failed to update StorageClusterPeer status.")
+	}
+
+	if reconcileErr != nil {
+		err = reconcileErr
+	} else if statusErr != nil {
+		err = statusErr
+	}
+
+	return result, err
+}
+
+func (r *StorageClusterPeerReconciler) reconcilePhases() (ctrl.Result, error) {
+	ocsClient, err := r.newExternalClusterClient()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	defer ocsClient.Close()
+
+	// marked for deletion
+	if !r.storageClusterPeer.GetDeletionTimestamp().IsZero() {
+		//TODO: Removing PeerOCS Call
+
+		if controllerutil.RemoveFinalizer(r.storageClusterPeer, storageClusterPeerFinalizer) {
+			r.log.Info("removing finalizer from StorageClusterPeer.", "StorageClusterPeer", r.storageClusterPeer.Name)
+			if err := r.update(r.storageClusterPeer); err != nil {
+				r.log.Info("Failed to remove finalizer from StorageClusterPeer", "StorageClusterPeer", r.storageClusterPeer.Name)
+				return reconcile.Result{}, fmt.Errorf("failed to remove finalizer from StorageClient: %v", err)
+			}
+		}
```

Comment:
Why do we have a finalizer if there are no clean-up steps implemented here?
If the intent is to have this as scaffolding for future de-peering steps I would say remove it an introduce it together with the de-peering code.